pkgs/development/tools/godot/default.nix

```diff
@@ -1,103 +1,146 @@
+
+    buildDescription = null;
   };
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ installShellFiles pkg-config scons ];
+
   buildInputs = [
-    scons
-    udev
+    alsa-lib
+    freetype
+    libGLU
+    libpulseaudio
     libX11
     libXcursor
+    libXext
+    libXfixes
+    libXi
     libXinerama
     libXrandr
     libXrender
-    libXi
-    libXext
-    libXfixes
-    freetype
     openssl
-    alsa-lib
-    libpulseaudio
-    libGLU
-    zlib
+    udev
     yasm
+    zlib
   ];
 
   patches = [ ./pkg_config_additions.patch ./dont_clobber_environment.patch ];
 
   enableParallelBuilding = true;
 
-  sconsFlags = "target=release_debug platform=x11";
-  preConfigure = ''
-    sconsFlags+=" ${
-      lib.concatStringsSep " "
-      (lib.mapAttrsToList (k: v: "${k}=${builtins.toJSON v}") options)
-    }"
-  '';
+  sconsFlags =
+    let
+      toSconsString = x: if builtins.isBool x then lib.boolToString x else toString x;
+    in
+      lib.mapAttrsToList (k: v: "${k}=${toSconsString v}") self.passthru.sconsFlagSet;
 
-  outputs = [ "out" "dev" "man" ];
+  shouldInstallBin = true;
+  shouldInstallShortcut = true;
+  shouldInstallHeaders = true;
+  shouldInstallManual = true;
+
+  outputs = []
+    ++ lib.optional (self.shouldInstallBin || self.shouldInstallShortcut) "out"
+    ++ lib.optional self.shouldInstallHeaders "dev"
+    ++ lib.optional self.shouldInstallManual "man";
+
+  # Where to install the godot bin files within the out directory.
+  binInstallPath = "bin";
+
+  # Pattern identifying the godot executable resulting from the scons build.
+  builtGodotBinNamePattern = "godot.*";
+
+  # Name to give the godot executable when installing in the out path.
+  installedGodotBinName = self.pname;
 
   installPhase = ''
-    mkdir -p "$out/bin"
-    cp bin/godot.* $out/bin/godot
+    runHook preInstall
+
+    if [ $shouldInstallBin ]; then
+      echo "Installing godot binaries."
+      bin="$out/$binInstallPath"
+      mkdir -p "$bin"
+      cp -R bin/. "$bin"
+      mv "$bin"/$builtGodotBinNamePattern "$bin/$installedGodotBinName"
+    fi
+
+    if [ $shouldInstallShortcut ]; then
+      echo "Installing godot shortcut."
+      share="$out"/share
+      mkdir -p "$share/"{applications,icons/hicolor/scalable/apps}
+      cp misc/dist/linux/org.godotengine.Godot.desktop "$share"/applications
+      cp icon.svg "$share"/icons/hicolor/scalable/apps/godot.svg
+      cp icon.png "$share"/icons/godot.png
+      substituteInPlace "$share"/applications/org.godotengine.Godot.desktop --replace "Exec=godot" "Exec=\"$out/bin/$pname\""
+    fi
 
-    mkdir "$dev"
-    cp -r modules/gdnative/include $dev
+    if [ $shouldInstallHeaders ]; then
+      echo "Installing godot headers."
+      mkdir -p "$dev"
+      cp -R modules/gdnative/include "$dev"
+    fi
 
-    mkdir -p "$man/share/man/man6"
-    cp misc/dist/linux/godot.6 "$man/share/man/man6/"
+    if [ $shouldInstallManual ]; then
+      echo "Installing godot manual."
```

Comment:
The man page is static and it will fail is the file does not exist. I don't think we need the extra verboseness here

Reply:
By that do you mean you don't think it should be optional?  The `shouldInstallManual` variable is currently overridden to false for the export-templates.nix derivation.  Of the 4 `shouldInstall` variables, the only one left true for the export-templates is `shouldInstallBin`.

The original derivation for `export-templates.nix`, before I made any changes, avoided installing the shortcut, the headers, and the manual when building export templates.  I'm not sure _why_ it was done, but I didn't want to change that.  This is why I originally factored the installation of binaries, shortcut, headers, and manual into self contained steps that could be explicitly listed by overriding derivations, and then changed into hard-coded steps toggled by booleans per your suggestion.

Reply:
I don't think we need an option if it is only used once. We should append it in the package to postInstall and remove this.